### PR TITLE
Issue 4414 - disk monitoring - prevent division by zero crash

### PR DESCRIPTION
**Pull request by mreynolds389**
Bug Description:  

If a disk mount has zero total space or zero used space then a division by zero can occur and the server will crash.

It has also been observed that sometimes a system can return the wrong disk entirely, and when that happens the incorrect disk also has zero available space which triggers the disk monitoring thread to immediately shut the server down.

Fix Description:  

Check the total and used space for zero and do not divide, just ignore it.  As a pre-emptive measure ignore disks from /dev, /proc, /sys (except /dev/shm). Yes it's a bit hacky, but the true underlying cause is not known yet.  So better to be safe than sorry.

Relates: https://github.com/389ds/389-ds-base/issues/4414

### CODE REVIEW

File: ldap/servers/slapd/daemon.c [hunk 1]
```diff
@@ -222,7 +222,27 @@ disk_mon_get_mount_point(char *dir)
         }
         if (s.st_dev == dev_id) {
             endmntent(fp);
-            return (slapi_ch_strdup(mnt->mnt_dir));
+            if ((strncmp(mnt->mnt_dir, "/dev", 4) == 0 ||
```
Comment (Firstyear) [inline ldap/servers/slapd/daemon.c]: I think the logic here looks wrong? Shouldn't it be:
```
if (
    (strncmp(mnt->mnt_dir, "/dev", 4) == 0 && strncmp(mnt->mnt_dir, "/dev/shm", 8) != 0) ||
    strncmp(mnt->mnt_dir, "/proc", 4) == 0 ||
    strncmp(mnt->mnt_dir, "/sys", 4) == 0)

```

**Review by Firstyear — CHANGES_REQUESTED** [review]
Beside that one logic comment, I think this is good.

File: ldap/servers/slapd/daemon.c (hunk 1 above)
Comment (mreynolds389) [reply ldap/servers/slapd/daemon.c]: You get the same result, but yours is clearer.

Comment (Firstyear) [reply ldap/servers/slapd/daemon.c]: Cool, lets go for clarity then, otherwise, ack :)

Comment (mreynolds389) [reply ldap/servers/slapd/daemon.c]: Agreed, changes made, and merging...

**Pull request merged**